src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AttendanceService.kt

```diff
@@ -34,29 +36,38 @@ class AttendanceService(
     createAttendanceRequest: CreateAttendanceRequest,
   ): CreateAttendanceResponse =
     attendanceRepository.save(createAttendanceRequest.toOffenderCourseAttendance(scheduleId, bookingId))
-      .let { CreateAttendanceResponse(it.eventId) }
+      .let { CreateAttendanceResponse(it.eventId, it.courseSchedule!!.courseScheduleId) }
 
-  private fun CreateAttendanceRequest.toOffenderCourseAttendance(scheduleId: Long, bookingId: Long): OffenderCourseAttendance {
-    val courseSchedule = scheduleRepository.findByIdOrNull(scheduleId)
-      ?: run { throw NotFoundException("Course schedule with id=$scheduleId not found") }
+  private fun CreateAttendanceRequest.toOffenderCourseAttendance(courseActivityId: Long, bookingId: Long): OffenderCourseAttendance {
```

Comment:
maybe this would read better not being an extension function, and just pass in the request given you need the repo looks up from the enclosing class?